scripts/production/homology_dump_cleanup.py

```diff
@@ -0,0 +1,265 @@
+        return user, host, port, database
+    else:
+        raise ValueError("Invalid database URL format")
+
+def mysql_connection(query, host, database, port, user):
+    """
+    Executes a given SQL query on a MySQL database and fetches the result.
+
+    This function establishes a connection to the MySQL database using provided connection details.
+    It then executes the given query using a cursor obtained from the connection. After executing the query,
+    it fetches all the rows of the query result and returns them. The function handles any errors that might
+    occur during the process and ensures that the database connection is closed before returning the result.
+
+    Args:
+        query (str): The SQL query to be executed.
+        database (str): The name of the database to connect to.
+        host (str): The host name or IP address of the MySQL server.
+        port (int): The port number to use for the connection.
+        user (str): The username to use for the database connection.
+
+    Returns:
+        tuple: A tuple of tuples containing the rows returned by the query execution.
+
+    Note:
+        This function does not handle database password authentication. Ensure that the provided user
+        has the necessary permissions and that the database is configured to allow password-less connections
+        from the given host.
+    """
+    try:
+        conn = pymysql.connect(
+            host=host, user=user, port=port, database=database.strip()
+        )
+        cursor = conn.cursor()
+        cursor.execute(query)
+        info = cursor.fetchall()
+    except pymysql.Error as err:
+        logging.error(f"MySQL Error: {err}")
+    cursor.close()
+    conn.close()
+    try: 
+        return info
+    except UnboundLocalError:
+        logging.error(f"\nNothing returned for SQL query: {query}\n")
+        sys.exit()
+
+def get_unique_collections(result_tuples):
+    """
+    Extracts and returns a sorted list of unique collection names from a list of tuples.
+
+    This function processes a list of tuples where each tuple contains a collection name as its first element.
+    If the collection name starts with the prefix 'collection-', this prefix is removed before adding the name
+    to a set to ensure uniqueness. Finally, the unique collection names are returned as a sorted list.
+
+    Args:
+    - result_tuples (list of tuples): A list of tuples, each containing collection names as their first element.
+
+    Returns:
+    - list: A sorted list of unique collection names with the 'collection-' prefix removed if present.
+
+    Example:
+    >>> get_unique_collections([('collection-abc',), ('collection-xyz',), ('abc',), ('collection-abc',)])
+    ['abc', 'xyz']
+    
+    The function uses a set to ensure uniqueness of the collection names. It then converts the set to a sorted list
+    before returning it.
+    """
+    unique_strings = set()  # Using a set to store unique strings
```

Comment:
Hi Botond, yes the set seems to be necessary here. Since the collections we get from the master database are prefixed with the word "collection-", when I remove that word we have some non-unique collections. I have used set to avoid that.